hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestOzonePrefixTable.java

```diff
@@ -0,0 +1,183 @@
+import org.apache.hadoop.ozone.om.helpers.OmPrefixInfo;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.Timeout;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.concurrent.TimeoutException;
+
+import static org.apache.hadoop.fs.CommonConfigurationKeysPublic.FS_TRASH_INTERVAL_KEY;
+import static org.apache.hadoop.ozone.OzoneConfigKeys.OZONE_FS_ITERATE_BATCH_SIZE;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.fail;
+
+/**
+ * Test verifies the entries and operations in prefix table.
+ */
+public class TestOzonePrefixTable {
+
+  @Rule
+  public Timeout timeout = new Timeout(300000);
+
+  private static final Logger LOG =
+          LoggerFactory.getLogger(TestOzoneFileSystem.class);
+
+  private MiniOzoneCluster cluster;
+  private FileSystem fs;
+  private OzoneFileSystem o3fs;
+  private String volumeName;
+  private String bucketName;
+  private int rootItemCount;
+
+  @Test(timeout = 300_000)
+  public void testMultiLevelDirs() throws Exception {
+    setupOzoneFileSystem();
+    // Op 1. create dir -> /d1/d2/d3/d4/
+    // Op 2. create dir -> /d1/d2/d3/d4/d5
+    // Op 3. create dir -> /d1/d2/d3/d4/d6
+    Path parent = new Path("/d1/d2/d3/d4/");
+    fs.mkdirs(parent);
+
+    OMMetadataManager omMgr = cluster.getOzoneManager().getMetadataManager();
+    OmBucketInfo omBucketInfo = omMgr.getBucketTable().get(
+            omMgr.getBucketKey(volumeName, bucketName));
+    Assert.assertNotNull("Failed to find bucketInfo", omBucketInfo);
+
+    ArrayList<String> prefixKeys = new ArrayList<>();
+    long d1ObjectID = verifyPrefixKey(omBucketInfo.getObjectID(), "d1", "/d1",
+            prefixKeys, omMgr);
+    long d2ObjectID = verifyPrefixKey(d1ObjectID, "d2", "/d1/d2", prefixKeys,
+            omMgr);
+    long d3ObjectID = verifyPrefixKey(d2ObjectID, "d3", "/d1/d2/d3",
+            prefixKeys, omMgr);
+    long d4ObjectID = verifyPrefixKey(d3ObjectID, "d4", "/d1/d2/d3/d4",
+            prefixKeys, omMgr);
+
+    // verify entries in prefix table
+    TableIterator<String, ? extends
+            Table.KeyValue<String, OmPrefixInfo>> iterator =
+            omMgr.getPrefixTable().iterator();
+    int count = prefixKeys.size();
+    Assert.assertEquals("Unexpected prefix table entries!", 4, count);
+    while (iterator.hasNext()) {
+      count--;
+      Table.KeyValue<String, OmPrefixInfo> value = iterator.next();
+      prefixKeys.remove(value.getKey());
+    }
+    Assert.assertEquals("Unexpected prefix table entries!", 0, count);
+
+    // verify entries in key table
+    TableIterator<String, ? extends
+            Table.KeyValue<String, OmKeyInfo>> keyTableItr =
+            omMgr.getKeyTable().iterator();
+    while (keyTableItr.hasNext()) {
+      fail("Shouldn't add any entries in KeyTable!");
```

Comment:
Can you add an additionally test cases for current entries in KeyTable? Now the new file key would be like objId/fileName rather than /volume/bucket/keyName.

Reply:
#verifyPrefixKey() function is doing the same by fetching entries using "omMgr.getPrefixTable().get(dbKey);" call.
Are you suggesting something different from this verification check?

Reply:
@rakeshadr , actually I mean we need to add logic to verify new entry on key table as well in utl. Current unit test only check the dir entry in prefix table.

Key table format before:
KeyName:
/vol1/buck1/a/b/c/d/file1

Now like this:
1028/file1

Reply:
Thanks for the details. I have created new PR https://github.com/apache/hadoop-ozone/pull/1404 and incorporated your comments. Please review it.